src/app/products/product-list/product-list.component.ts

```diff
@@ -59,19 +80,23 @@ export class ProductListComponent implements OnInit {
     {
       headerName: 'Title',
       field: 'title',
+      filter: 'title',
```

Comment:
Missed the part that filters should be implemented on server side.
There should be an additional call in a service, which will be triggered by form control.
Once the data have come we should update product behaviour subject.